File: eventorganizer/frontend/event-frontend/src/app/core/model/Event.ts
// event.model.ts

export enum EventType {
  CORPORATE = "CORPORATE",
  WEDDING = "WEDDING",
  COMMUNITY_FESTIVAL = "COMMUNITY_FESTIVAL",
  BIRTHDAY = "BIRTHDAY",
  CONCERT = "CONCERT",
  SPORTS = "SPORTS",
  CHARITY = "CHARITY",
}

export enum Status {
  PENDING = 'PENDING',
  IN_PROGRESS = 'IN_PROGRESS',
  COMPLETED = 'COMPLETED',
  CANCELLED = 'CANCELLED',
}

export interface IEvent {
  id?: number;
  userId: number | null;
  name: string;
  title: string;
  location: string;
  description: string;
  estimatedExpense: number | null;
  actualExpense: number | null;
  startDate: string;
  endDate: string;
  type: EventType;
  status: Status;
  // clientId: number | null;
  vendors: number[];  // Change from string[] to number[]
  guests: number[];   // Change from string[] to number[]
}

export interface EventRequest {
  id: number;
  eventId: number;
  vendorId: number;
  status: IStatus;//need to change here
}

export enum IStatus {
  APPROVED = 'APPROVED',
  REJECTED = 'REJECTED',
  PENDING = 'PENDING',
}

File: eventorganizer/Backend/event-service/src/main/java/com/eventManagement/model/Event.java
package com.eventManagement.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;


@Data
@Entity
public class Event
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long userId;
    private String name;
    private String title;
    private String location;
    private String description;
    private Double estimatedExpense;
    private Double actualExpense;

    //    @Temporal(TemporalType.TIMESTAMP)
//    @DateTimeFormat(style = "dd-MM-yyyy hh:mm")
//    @JsonFormat(shape=JsonFormat.Shape.STRING, pattern="dd-MM-yyyy hh:mm")
    private LocalDate startDate;
    private LocalDate endDate;
    private EventType type;
    private Status status;

//    private Long clientId;

    @ElementCollection(fetch = FetchType.EAGER)
            @JsonProperty("vendorIds")
    Set<Long> vendors=new HashSet<>();

    @ElementCollection(fetch = FetchType.EAGER)
    List<Long>guests=new ArrayList<>();

//    @ElementCollection
//    private List<Long> clientIds = new ArrayList<>();

//
//    @Transient
//    private double totalValue;


}
